redis/src/tls.rs

```diff
@@ -0,0 +1,146 @@
+use std::io::{BufRead, Cursor, Error, ErrorKind as IOErrorKind};
+
+use rustls::{Certificate, OwnedTrustAnchor, PrivateKey, RootCertStore};
+
+use crate::{Client, ConnectionAddr, ErrorKind, IntoConnectionInfo, RedisError, RedisResult};
+
+/// Structure to hold mTLS client _certificate_ and _key_ binaries in PEM format
+///
+#[derive(Clone)]
+pub struct ClientTlsBinary {
+    /// client certificate byte stream in PEM format
+    pub client_cert: Vec<u8>,
+    /// client key byte stream in PEM format
+    pub client_key: Vec<u8>,
+}
+
+/// Structure to hold TLS certificates
+/// - `client_tls`: binaries of clientkey and certificate within a `ClientTlsBinary` structure if mTLS is used
+/// - `root_cert`: binary CA certificate in PEM format if CA is not in local truststore
+///
+#[derive(Clone)]
+pub struct CertificatesBinary {
+    /// 'ClientTlsBinary' containing client certificate and key if mTLS is to be used
+    pub client_tls: Option<ClientTlsBinary>,
+    /// root certificate byte stream in PEM format if the local truststore is *not* to be used
+    pub root_cert: Option<Vec<u8>>,
+}
```

Comment:
Not very important, but why the `Binary` in the name ? It kinda alludes to DER encoding in this situation, which isn't the case here, since we're using PEM.

Reply:
I thought of `Binary` cos it contains just the binary streams for keys and certificates, and to make clear that's not a PEM validated structure yet.
... but if it's confusing maybe `Raw` would work better?

Reply:
I think @thalesfragoso makes a fair point. What about just `ClientTlsConfig` and `TlsCertificates` instead of  ClientTlsBinary and CertificatesBinary, respectively?